Lib/ldap/extop/__init__.py

```diff
@@ -38,19 +46,159 @@ def encodedRequestValue(self):
     return self.requestValue
 
 
-class ExtendedResponse:
+class ExtendedResponse(ldap.response.ExtendedResult):
   """
   Generic base class for a LDAPv3 extended operation response
 
-  requestName
-      OID as string of the LDAPv3 extended operation response
+  responseName
+      OID as string of the LDAPv3 extended operation response or None
   encodedResponseValue
       BER-encoded ASN.1 value of the LDAPv3 extended operation response
   """
 
-  def __init__(self,responseName,encodedResponseValue):
+  def __init_subclass__(cls):
+    if not getattr(cls, 'responseName', None):
+      return
+
+    KNOWN_EXTENDED_RESPONSES.setdefault(cls.responseName, cls)
+
+  @classmethod
+  def __convert_old_api(cls, responseName_or_msgid=_NOTSET,
+                        encodedResponseValue_or_msgtype=_NOTSET,
+                        controls=None, *,
+                        result=_NOTSET, matcheddn=_NOTSET, message=_NOTSET,
+                        referrals=_NOTSET, name=None, value=None,
+                        defaultClass: Optional[type['ExtendedResult']] = None,
+                        msgid=_NOTSET, msgtype=_NOTSET,
+                        responseName=_NOTSET, encodedResponseValue=_NOTSET,
+                        **kwargs):
+    """
+    Implements both old and new API:
+    __init__(self, responseName, encodedResponseValue)
+    and
+    __init__/__new__(self, msgid, msgtype, controls=None, *,
+                     result, matcheddn, message, referrals,
+                     defaultClass=None, **kwargs)
+    """
```

Comment:
It looks overly complex, IMO.
Can we split it into two?

Something like
```python
@classmethod
def from_old_api(cls, responseName, encodedResponseValue)
@classmethod  
def from_new_api(cls, msgid, msgtype, ...)
```

Reply:
(Ah found it, there's several copies of this comment in the interface, copying my reply here)

I'm happy with anything of the sort as long as there's a way to reuse most of the decoding and registration code in `ldap.extop`. If we can find a way that gets rid of this as well, I'll only be happier, not less so.